pkg/controller/scylladbdatacenter/resource.go

```diff
@@ -2039,6 +2055,34 @@ func applyRackTemplateOnRackSpec(rackTemplate *scyllav1alpha1.RackTemplate, rack
 					},
 				}
 			}(),
+			ExposeOptions: func() *scyllav1alpha1.RackExposeOptions {
+				if rackTemplate.ExposeOptions == nil && rack.ExposeOptions == nil {
+					return nil
+				}
+
+				rackExposeOptions := &scyllav1alpha1.RackExposeOptions{}
+				for _, reo := range []*scyllav1alpha1.RackExposeOptions{rackTemplate.ExposeOptions, rack.ExposeOptions} {
+					if reo == nil {
+						continue
+					}
```

Comment:
This `if` and the below `if` express very similar things but with the logic inverted.

How about:
```go
if reo == nil || reo.NodeService == nil {
    continue
}
```

Reply:
It's intended, as global ExposeOptions currently have more options than NodeService. Extending the code in case more rack-level expose options would be available, wouldn't require changing `if` you're proposing, and is more explicit about the expected flow. Leaving as is.